src/device/index.js

```diff
@@ -1182,65 +1187,87 @@ realityEditor.device.onDocumentMultiTouchMove = function (event) {
+                console.log('center: ', centerTouch);
+                console.log('outer: ', outerTouch);
+                realityEditor.gui.ar.positioning.scaleVehicle(activeVehicle, centerTouch, outerTouch);
 
             } else {
 
-                // if you have two fingers on the screen (one on the frame, one on the canvas)
-                // make sure the scale event is centered around the frame
-                [].slice.call(event.touches).forEach(function(touch){
-
-                    let targetId = realityEditor.device.utilities.getVehicleIdFromTargetId(touch.target.id);
-                    var didTouchOnFrame = targetId === activeVehicle.uuid;
-                    var didTouchOnNode = targetId === activeVehicle.frameId + activeVehicle.name;
-                    var didTouchOnPocketContainer = touch.target.className === "element-template";
-                    if (didTouchOnFrame || didTouchOnNode || didTouchOnPocketContainer) {
-                        centerTouch = {
-                            x: touch.pageX,
-                            y: touch.pageY
-                        };
-                    } else {
-                        outerTouch = {
-                            x: touch.pageX,
-                            y: touch.pageY
-                        };
+                // consider a touch on 'object__frameKey__' and 'svgobject__frameKey__' to be on the same target
+                // also consider a touch that started on pocket-element to be on the frame element
+                var touchTargets = Array.from(event.touches).map(function(touch) {
+                    var targetId = realityEditor.device.utilities.getVehicleIdFromTargetId(touch.target.id);
+                    if (targetId === 'pocket-element') {
+                        targetId = activeVehicle.uuid;
                     }
+                    return targetId;
                 });
 
-            }
-
-            realityEditor.gui.ar.positioning.scaleVehicle(activeVehicle, centerTouch, outerTouch);
+                var areBothOnElement = touchTargets[0] === touchTargets[1];
+
+                var centerTouch;
+                var outerTouch;
+
+                if (areBothOnElement) {
+                    // if you do a pinch gesture with both fingers on the frame
+                    // center the scale event around the first touch the user made
+                    centerTouch = {
+                        x: event.touches[0].pageX,
+                        y: event.touches[0].pageY
+                    };
+                    outerTouch = {
+                        x: event.touches[1].pageX,
+                        y: event.touches[1].pageY
+                    };
+                } else {
+                    // if you have two fingers on the screen (one on the frame, one on the canvas)
+                    // make sure the scale event is centered around the frame
+                    [].slice.call(event.touches).forEach(function(touch){
```

Comment:
Array.from() is more standard than [].slice.call

Reply:
Thanks for the tip! I sprinkled a few of those in awhile ago – I'll go ahead and swap them all out with Array.from while I'm at it